lib/features/email_recovery/presentation/email_recovery_controller.dart

```diff
@@ -338,6 +342,35 @@ class EmailRecoveryController extends BaseController with DateRangePickerMixin {
     popBack();
   }
 
+  String getRestorationHorizonAsString() {
+    return ((arguments!.session
+        .props[0] as Map<CapabilityIdentifier, CapabilityProperties>?)
+        ?[capabilityDeletedMessagesVault]
+        ?.props[0] as Map<String, dynamic>?)
+        ?['restorationHorizon']
+        ?? "15 days";
+  }
+
+  Duration getRestorationHorizonAsDuration() {
+    String horizonWithCorrectFormat = getRestorationHorizonAsString()
+        .replaceAll(" days", "d")
+        .replaceAll(" day", "d")
+        .replaceAll(" hours", "h")
+        .replaceAll(" hour", "h")
+        .replaceAll(" minutes", "m")
+        .replaceAll(" minute", "m")
+        .replaceAll(" seconds", "s")
+        .replaceAll(" second", "s")
+        .replaceAll(" milliseconds", "ms")
+        .replaceAll(" millisecond", "ms");
+
+    return parseDuration(horizonWithCorrectFormat, separator: ' ');
```

Comment:
If the whole purpose of `duration` package is to use `parseDuration`, please write your own `parse` function and remove the package from `pubspec.yaml`. Also unit test that function.

Reply:
i copied the `parseDuration` function from https://github.com/desktop-dart/duration/blob/master/lib/src/parse/parse.dart#L6, should i include the license ?